docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1128 @@
 JSON functions and operators
 ============================
 
+The SQL standard describes functions and operators to process JSON data. They
+allow you to access JSON data according to its structure, generate JSON data,
+and store it persistently in SQL tables.
+
+Importantly, the SQL standard imposes that there be no dedicated data type to
+represent JSON data in SQL. Instead, JSON data shall be represented as
+character or binary strings. Although Trino supports ``JSON`` type, it is not
+used or produced by the following functions.
+
+Trino supports three functions for querying JSON data:
+:ref:`JSON_EXISTS<json_exists>`,
+:ref:`JSON_QUERY<json_query>`, and :ref:`JSON_VALUE<json_value>`. Each of them
+is based on the same mechanism of exploring and processing JSON input using
+JSON path.
+
+JSON path language
+------------------
+
+The JSON path language is a special language, used exclusively by certain SQL
+operators to specify the query to perform on the JSON input. Although the path
+expression is embedded in an SQL query, the syntax of the path language differs
+from SQL. However, the semantics of predicates, operators, etc. in the JSON
+path generally follow SQL. Also, the path language is case-sensitive both in
+keywords and identifiers.
+
+.. _path_syntax_and_semantics:
+
+JSON path syntax and semantics
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+JSON path expression, similarly to SQL expression, is a recursive structure.
+Although the name "path" suggests a linear sequence of operations going step by
+step deeper into the JSON structure, the JSON path is in fact a tree. It can
+access the input JSON item multiple times, in multiple ways, and combine the
+results. Moreover, the result of the JSON path is not a single item, but an
+ordered sequence of items. Each of the sub-expressions takes one or more input
+sequences and returns a sequence as the result.
+
+Note: in the lax mode, most path operations first unnest all JSON arrays in the
+input sequence. Any divergence from this rule is mentioned in the following
+listing. Path modes are explained in :ref:`path_modes`
+
+The JSON path language supports the following features:
+
+**literals**
+
+- numeric literals
```

Comment:
Does this all render okay in terms of indentation?

Reply:
I think so. And after switching `**literals**` to a subsection it renders exactly the same.